core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -276,6 +282,37 @@ export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
     });
   }
 
+  getCheckpointsCardIndexs(): number[] {
+    const checkpointCardIndexs: number[] = [];
+    const numberOfCards = this.playerTranscriptService.getNumCards();
+
+    for (let i = 0; i < numberOfCards; i++) {
+      const stateName = this.playerTranscriptService.getCard(i).getStateName();
+      const correspondingState =
+        this.explorationEngineService.getStateFromStateName(stateName);
+      if (correspondingState.cardIsCheckpoint) {
+        checkpointCardIndexs.push(i);
+      }
+    }
+    return checkpointCardIndexs;
+  }
+
+  returnToCheckpointIfCompleted(index: number): void {
```

Comment:
Maybe you can give "index" a more descriptive name, to distinguish it from the values in checkpointCardsIndex. Is it the checkpointNumber?

Reply:
I changed the name accordingly